drivers/gpio/gpio_gd32.c

```diff
@@ -46,7 +46,9 @@ struct gpio_gd32_config {
 	struct gpio_driver_config common;
 	uint32_t reg;
 	uint16_t clkid;
+#ifdef CONFIG_GD32_EXTI
```

Comment:
It looks as though this will be padded out to 32-bits in any case. If you remove the `#ifdef` here, then you can switch to using `if (IS_ENABLED(CONFIG_GD32_EXTI)` elsewhere and remove the `#ifdef` around `static void gpio_gd32_isr(uint8_t line, void *arg)`

Reply:
> It looks as though `clkid` will be padded out to 32-bits if `CONFIG_GD32_EXTI=n`, so the `#ifdef` can be removed here.

Insightful!
Removed the `#ifdef`.



> Then you can switch to using `if (IS_ENABLED(CONFIG_GD32_EXTI)` elsewhere and remove the `#ifdef` around `static void gpio_gd32_isr(uint8_t line, void *arg)`

Commenting in follows topics.